web3.js/src/connection.ts

```diff
@@ -2945,9 +2959,46 @@ export class Connection {
           () => resolve({__type: TransactionStatus.TIMED_OUT, timeoutMs}),
           timeoutMs,
         );
+      } else if (Object.prototype.hasOwnProperty.call(strategy, 'nonceHash')) {
+        const {nonceAccount, nonceHash} =
+          strategy as NonceBasedTransactionConfirmationStrategy;
+        let currentNonceHash = nonceHash;
+        const getCurrentNonceHash = async () => {
+          try {
+            const nonceAccountData = await this.getNonce(
+              nonceAccount,
+              commitment,
+            );
+            return nonceAccountData!.nonce;
```

Comment:
It's not obvious that this could throw an error if a nonce account wasn't found. If the nonce isn't found (is `null`) then `NONCE_INVALID` should be bubbled up to the user

Reply:
Hmm… I did this _sort_ of on purpose. This code gets caught by a block whose MO is to use the last-known value of the nonce if ever you can't fetch the nonce. The idea here was that expiring the transaction when you're not sure is worse than hoping that the signature will confirm at some point.

I'll rethink this.